packages/cdktf-cli/lib/get/generator/models/resource-model.ts

```diff
@@ -91,6 +92,14 @@ export class ResourceModel {
     return this.isProvider ? this.provider : this.isDataSource ? this.terraformType.replace(/^data_/, '') : this.terraformType
   }
 
+  public get terraformDocName(): string {
+    if (this.isDataSource) {
+      return toSnakeCase(this.className.replace(new RegExp(`^data${this.provider}`, 'i'), ''))
```

Comment:
If the name collision code hits, then the class name will no longer match. `terraformResourceType` + removing the provider might be a better match for this case.

Reply:
thanks for the feedback - caaf6ca addressed this 👍